pkg/reconciler/trigger/trigger.go

```diff
@@ -136,16 +144,41 @@ func (r *Reconciler) reconcile(ctx context.Context, t *v1alpha1.Trigger) error {
 		// Everything is cleaned up by the garbage collector.
 		return nil
 	}
+	track := r.resourceTracker.TrackInNamespace(t)
+	dependencyAnnotation, ok := t.GetAnnotations()[v1alpha1.DependencyAnnotation]
+	if ok {
+		dependencyObjRef, err := getObjRefFromDependencyAnnotation(dependencyAnnotation)
+		if err != nil {
+			logging.FromContext(ctx).Error("Unable to unmarshal objectReference from dependency annotation of trigger:", zap.Error(err))
+			return err
+		}
+		if err := validateDependencyAnnotation(dependencyObjRef); err != nil {
+			logging.FromContext(ctx).Error("Failed to validate dependency annotation of trigger", zap.Error(err))
+			t.Status.MarkDependencyReadyFailed("InvalidDependencyAnnotation", "Failed to validate dependency annotation of trigger")
+			return err
+		}
+		//Assume that trigger and its dependency importer are in the same namespace
+		if err := track(dependencyObjRef); err != nil {
+			logging.FromContext(ctx).Error("Unable to track changes to Dependency", zap.Error(err))
+			return err
+		}
+		e := r.triggerMarkedReadyByDependency(dependencyObjRef, ctx, t)
+		if e != nil {
+			logging.FromContext(ctx).Error("Unable to reconcile dependency", zap.Error(err))
+			return e
```

Comment:
```suggestion
			return fmt.Errorf("propagating dependency readiness: %v", err)
```

And remove the log line above.